openquake/hazardlib/geo/surface/kite_fault.py

```diff
@@ -263,8 +295,11 @@ def get_width(self) -> float:
         return self.width
 
     def get_dip(self) -> float:
+        # TODO this method is provisional. It works correctly for simple and
+        # regular geometries defined using profiles parallel to the dip
+        # direction
```

Comment:
Can we raise an error when the method does not work correctly?

Reply:
We cannot because if I were able to create such a test I would have also fixed the method :) This is the best we can do at present. In a following PR, I will work at improving this.